client/src/components/Header/Header.tsx

```diff
@@ -14,42 +16,80 @@ const Header: React.FC = () => {
 
   const navigate = useNavigate();
-
+  const { scrollPosition, isScrolledUp } = useScroll();
   const [isActive, setIsActive] = useState(false);
+  // * const [isHovered, setIsHovered] = useState(false);
+  const isTopBarVisible = scrollPosition < 50;
+  const shouldShowTopBar = isTopBarVisible || isScrolledUp;
+  // * const shouldShowTopBar = isTopBarVisible || isScrolledUp || isHovered;
 
   /* -------------------------------------------------------------------------- */
   /*                                  Function                                  */
   /* -------------------------------------------------------------------------- */
 
   const toggleHamburger = () => setIsActive(!isActive);
 
+  // ? J'ai implémenté une fonctionnalité qui permet d'afficher la topbar au survol du header, mais je suis pas sûr que ce soit une bonne idée.
+  // ? Pour la tester, il vous suffit de décommenter les lignes commençant par "// *".
+
+  // Handlers pour le survol
+  // * const handleMouseEnter = () => setIsHovered(true);
+  // * const handleMouseLeave = () => setIsHovered(false);
+
   /* -------------------------------------------------------------------------- */
   /*                                   Render                                   */
   /* -------------------------------------------------------------------------- */
 
   return (
-    <header className={styles.header}>
+    <header
+      className={classNames(styles.header, {
+        [styles.compactMode]: !shouldShowTopBar,
+      })}
+      // * onMouseEnter={handleMouseEnter}
+      // * onMouseLeave={handleMouseLeave}
+    >
       {/* Top banner */}
-      <div className={styles.topBanner}>
-        <button className={styles.helpButton}>
-          <img
-            src={helpIcon}
-            alt="Besoin d'aide ?"
-            aria-label="Besoin d'aide ?"
-            loading="lazy"
-          />
-          Besoin d’aide ?
-        </button>
-        <button className={styles.locationButton}>
-          <img
-            src={locationIcon}
-            alt="Ajouter ma localisation"
-            aria-label="Ajouter ma localisation"
-            loading="lazy"
-          />
-          Ajouter ma localisation
-        </button>
-      </div>
+      <motion.div
+        initial={{ opacity: 0, paddingTop: 0, paddingBottom: 0, maxHeight: 0 }}
+        animate={{
+          opacity: shouldShowTopBar ? 1 : 0,
+          paddingTop: shouldShowTopBar ? "6px" : "0",
+          paddingBottom: shouldShowTopBar ? "6px" : "0",
+          maxHeight: shouldShowTopBar ? "100px" : "0",
+        }}
+        className={styles.topBanner}
+      >
+        <div className={styles.leftContainer}>
+          <button type="button" className={styles.button}>
+            <MapPin size={16} className={styles.icon} />
+            <span>Paris 11e</span>
+          </button>
+
+          <div className={styles.separator}></div>
+
+          <button type="button" className={styles.button}>
```

Comment:
Tu mettrais quoi sur cette partie ?

Reply:
> Tu mettrais quoi sur cette partie ?

Ça peut être polyvalent, mettre par exemple en avant des produits de saison, des spécialités de régions, des événements spécifiques... sous forme de modal, peut-être

Reply:
Ok why not, ça me va. Faudra juste qu'on pense à créer des tickets pour l'implémentation de la feature (front & back)